sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -63,6 +64,45 @@
   // No instance of this class.
   private Samplers() {}
 
+  /**
+   * Returns a {@link Decision} with the given {@code attributes} and {@link Decision#isSampled()}
+   * returning {@code isSampled}.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
+   *
+   * <p>Using {@link #emptyDecision(boolean)} instead of this method is slightly faster and shorter
+   * if you don't need attributes.
+   *
+   * @param isSampled The value to return from {@link Decision#isSampled()}.
+   * @param attributes The attributes to return from {@link Decision#getAttributes()}. A different
+   *     object instance with the same elements may be returned. The map must not be modified after
+   *     being passed to this function. Use {@link Collections#emptyMap()} for an empty decision.
```

Comment:
Just having this note in here, rather than copying the attributes, feels too weak. If this is truly a hard requirement, the API should protect itself, rather than relying on implementors to read this javadoc.

Reply:
First, I think this is an SDK-only API that will only be used by sampler implementations. An "advanced" API so to speak. If I knew that this function was uselessly copying my map, I would rather write my own replacement for efficiency.

Second, in the currently only usage of the Sampler the attributes are actually copied into the span right-away, so I think ignored this note won't actually do any harm.

Reply:
Then the word "must" here seems overly strong, and possibly misleading. Can you update the javadoc to reflect the reality of the situation?

Reply:
Hmm, I wouldn't place too much value on the second point, as the usage of the sampler may change in the future. So anyone who reads the javadoc certainly MUST NOT modify this later.

Reply:
All I'm saying is: If this is an important constraint, then we should either a) enforce it via code or b) clearly explain what the ramifications of violating it are, and not just mention it offhand in attribute documentation.

Reply:
Unfortunately I think a) is not possible without losing efficiency. And I think with b) we would over-constrain ourselves regarding future changes.

Reply:
How about adding a line to the javadoc, not in the attributes section like this:

```
You should not modify the contents of the attributes after passing them to this function. If you modify the contents of the attributes after passing them to this function, behavior is not well defined.
```

Reply:
> a) is not possible without losing efficiency.

Confused about this, you should cache the result if performance is important, otherwise it is good to have your own implementation of the `Decision` where you ensure the immutable part differently.

Having an API that relies on these kind of comments is not safe and we cannot guarantee correctness (we can NLP or other exceptions if the user changes the map while we are reading it).

Reply:
> you should cache the result if performance is important

Good point actually. I think you convinced me that it is better to copy the attributes after all. I will add this to `Decision.create`.

Reply:
Waiting for this :)

Reply:
Done! I used `ImmutableMap.copyOf` and also fixed the javadoc.